lti_consumer/lti_1p3/consumer.py

```diff
@@ -90,20 +93,30 @@ def _get_user_roles(role):
 
     def prepare_preflight_url(
             self,
-            hint="hint",
-            lti_hint="lti_hint"
+            launch_data,
     ):
         """
         Generates OIDC url with parameters
         """
+        user_id = launch_data.user_id
+
+        # Set the launch_data in the cache. We store the launch_data in the cache instead of sending a representation of
+        # the launch_data (e.g. a JWT) as a query or form parameter because we cannot guarantee that the length of the
+        # GET query parameter will not exceed some browser maximum length and we need to support GET and POST requests
+        # back from tools.
+        # TODO: What is the expiration value by default? Will it provide enough time for the learner to get back?
```

Comment:
I think we should explicitly set this instead of relying on some internal default just so it's more visible. We can set this pretty short if we wanted but even 60s is a lot of buffer for an auth callback.

Reply:
That's fair. I'm not concerned as much about the authentication callback for learners on the LMS where the LTI launch is in an iframe, because the LTI launch occurs automatically after the page is rendered.

However, because the launch data is cached when the XBlock renders and the preflight URL is generated, this poses a problem for delayed LTI launches. This includes learner LMS LTI launches that are configured to launch in a modal or new window and Studio deep linking LTI launches for course staff (initiated by clicking a URL). I'm concerned that the data will not be cached long enough these LTI launches if we choose a 60 second timeout. Course staff or learners may leave pages open for longer than this and reasonably expect them to be operational.

The default timeout is 5 minutes. This is the same timeout for the JWT ID token. I don't see a harm in caching these values for this long to make launches more user friendly. If you'd like, I can include `django.core.cache.backends.base.DEFAULT_TIMEOUT` as an an argument to `TieredCache.set_all_tiers` to make clear the timeout.

What do you think?

I'm also thinking about providing more specific instructions on the error page in cases where the cache has expired, as opposed to our generic, "contact the course team" message. If it's as simple as reloading the LMS or Studio page, we should instruct them to do so instead of sending them to support.

Reply:
oh interesting I hadn't considered the fact that some of these preflight url's just hang around before getting consumed. Maybe that's something we could improve on (but later) I'd think a simple message to reload would be sufficient.

In any case I also don't see any issues with holding these for 5mins given the scale of use on edx.org (assuming all of that moved to 1.3 some day). But, why not just define our own timeout value to pass in instead, or is that default a platform setting we want to be configurable?

Reply:
I didn't see a need to specify a value that's also the default, but I understand that it'll be more readable to explicitly specify the timeout we want. I'll add that in. 

I'm going to add in some logging to capture cache misses. It will be good to keep an eye on how frequently those occur in case 5 minutes is not long enough.